include/picongpu/fields/currentDeposition/Cache.hpp

```diff
@@ -46,47 +46,38 @@ namespace picongpu
                  *
                  * @attention thread-collective operation, requires external thread synchronization
                  */
-                template<uint32_t T_numWorkers, typename T_BlockDescription, typename T_Acc, typename T_FieldBox>
-                DINLINE static auto create(T_Acc const& acc, T_FieldBox const& fieldBox, uint32_t const workerIdx)
+                template<typename T_BlockDescription, typename T_Worker, typename T_FieldBox>
+                DINLINE static auto create(T_Worker const& worker, T_FieldBox const& fieldBox)
 #if(!BOOST_COMP_CLANG)
                     -> decltype(CachedBox::create<0u, typename T_FieldBox::ValueType>(
-                        acc,
+                        worker,
                         std::declval<T_BlockDescription>()))
 #endif
                 {
                     using ValueType = typename T_FieldBox::ValueType;
                     /* this memory is used by all virtual blocks */
-                    auto cache = CachedBox::create<0u, ValueType>(acc, T_BlockDescription{});
+                    auto cache = CachedBox::create<0u, ValueType>(worker, T_BlockDescription{});
 
                     Set<ValueType> set(ValueType::create(0.0_X));
-                    ThreadCollective<T_BlockDescription, T_numWorkers> collectiveFill(workerIdx);
+                    auto collectiveFill = makeThreadCollective<T_BlockDescription>();
 
                     /* initialize shared memory with zeros */
-                    collectiveFill(acc, set, cache);
+                    collectiveFill(worker, set, cache);
                     return cache;
                 }
 
                 /** Flush the cache
                  *
                  * @attention thread-collective operation, requires external thread synchronization
                  */
-                template<
-                    uint32_t T_numWorkers,
-                    typename T_BlockDescription,
-                    typename T_Acc,
-                    typename T_FieldBox,
-                    typename T_FieldCache>
-                DINLINE static void flush(
-                    T_Acc const& acc,
-                    T_FieldBox fieldBox,
-                    T_FieldCache const& cachedBox,
-                    uint32_t const workerIdx)
+                template<typename T_BlockDescription, typename T_Worker, typename T_FieldBox, typename T_FieldCache>
+                DINLINE static void flush(T_Worker const& worker, T_FieldBox fieldBox, T_FieldCache const& cachedBox)
                 {
                     typename T_Strategy::GridReductionOp const op;
-                    ThreadCollective<T_BlockDescription, T_numWorkers> collectiveAdd(workerIdx);
+                    auto collectiveAdd = makeThreadCollective<T_BlockDescription>();
```

Comment:
Why is the functor instance called `collectiveAdd`? The operation `op` is supplied by the template parameter `T_Strategy` and the use type name does not suggest anything beyond, that the operation is a reduction. The documentation of this class also does not specify, that the reduction mus be an addition. Is this a misnomer?

Reply:
Can we remove the `Add` the? Or maybe replace it be `Reduction`. This is unrelated to this PR. But due to the small size of this change, it would never change otherwise, so this seems to be a good opportunity to fix it.

Reply:
It looks like the `Add` is still here...

Reply:
changed to `collectiveFlush`